lua/wikis/commons/Widget/NavBox/Child.lua

```diff
@@ -0,0 +1,147 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Widget/NavBox/Child
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Lua = require('Module:Lua')
+
+local Array = Lua.import('Module:Array')
+local Class = Lua.import('Module:Class')
+local FnUtil = Lua.import('Module:FnUtil')
+local Image = Lua.import('Module:Image')
+local Json = Lua.import('Module:Json')
+local Logic = Lua.import('Module:Logic')
+
+local Widget = Lua.import('Module:Widget')
+local WidgetUtil = Lua.import('Module:Widget/Util')
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local Div = HtmlWidgets.Div
+local Tbl = HtmlWidgets.Table
+local Tr = HtmlWidgets.Tr
+local Th = HtmlWidgets.Th
+local Td = HtmlWidgets.Td
+
+local NavBoxList = Lua.import('Module:Widget/NavBox/List')
+
+local EMPTY_CHILD_ERROR = 'Empty child found'
+
+---@class NavBoxChild: Widget
+---@operator call(table): NavBoxChild
+local NavBoxChild = Class.new(Widget)
+NavBoxChild.defaultProps = {
+	imagesize = '30px',
+	imagelink = '',
+	imageleftsize = '30px',
+	imageleftlink = '',
+}
+
+---@return Widget?
+function NavBoxChild:render()
+	local props = self.props
+
+	assert(props[1] or props.child1, EMPTY_CHILD_ERROR)
+	if not props.child1 then
+		return NavBoxList{children = Array.mapIndexes(function(index)
+			return self.props[index]
+		end)}
+	end
+
+	local children = Array.mapIndexes(function(rowIndex)
+		return NavBoxChild._getChild(props['child' .. rowIndex])
+	end)
+
+	if props[1] then
+		table.insert(children, {name = props.name, child = NavBoxList(props)})
+	end
```

Comment:
The input is modified for the frist NavBoxList but not the second, is this intended?